patches/cpp/VariableDatum.cpp.patch

```diff
@@ -1,150 +1,150 @@
--// AS IS AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
--// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
--// FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
--// COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
--// INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
--// BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
--// LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
--// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
--// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
--// ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
--// POSSIBILITY OF SUCH DAMAGE.
+ using namespace DIS;
+-
+-
+-VariableDatum::VariableDatum():
+-   _variableDatumID(0), 
+-   _variableDatumLength(0), 
+-   _variableDatums(0)
+-{
+-}
+-
+-VariableDatum::~VariableDatum()
+-{
+-    _variableDatums.clear();
+-}
+-
+-unsigned int VariableDatum::getVariableDatumID() const
+-{
+-    return _variableDatumID;
+-}
+-
+-void VariableDatum::setVariableDatumID(unsigned int pX)
+-{
+-    _variableDatumID = pX;
+-}
+-
+-unsigned int VariableDatum::getVariableDatumLength() const
+-{
+-   return _variableDatums.size();
+-}
+-
+-std::vector<EightByteChunk>& VariableDatum::getVariableDatums() 
+-{
+-    return _variableDatums;
+-}
+-
+-const std::vector<EightByteChunk>& VariableDatum::getVariableDatums() const
+-{
+-    return _variableDatums;
+-}
+-
+-void VariableDatum::setVariableDatums(const std::vector<EightByteChunk>& pX)
+-{
+-     _variableDatums = pX;
+-}
+-
+-void VariableDatum::marshal(DataStream& dataStream) const
+-{
+-    dataStream << _variableDatumID;
+-    dataStream << ( unsigned int )_variableDatums.size();
+-
+-     for(size_t idx = 0; idx < _variableDatums.size(); idx++)
+-     {
+-        EightByteChunk x = _variableDatums[idx];
+-        x.marshal(dataStream);
+-     }
+-
+-}
+-
+-void VariableDatum::unmarshal(DataStream& dataStream)
+-{
+-    dataStream >> _variableDatumID;
+-    dataStream >> _variableDatumLength;
+-
+-     _variableDatums.clear();
+-     for(size_t idx = 0; idx < _variableDatumLength; idx++)
+-     {
+-        EightByteChunk x;
+-        x.unmarshal(dataStream);
+-        _variableDatums.push_back(x);
+-     }
+-}
+-
+-
+-bool VariableDatum::operator ==(const VariableDatum& rhs) const
+- {
+-     bool ivarsEqual = true;
+-
+-     if( ! (_variableDatumID == rhs._variableDatumID) ) ivarsEqual = false;
+-
+-     for(size_t idx = 0; idx < _variableDatums.size(); idx++)
+-     {
+-        if( ! ( _variableDatums[idx] == rhs._variableDatums[idx]) ) ivarsEqual = false;
+-     }
+-
+-
+-    return ivarsEqual;
+- }
+-
+-int VariableDatum::getMarshalledSize() const
+-{
+-   int marshalSize = 0;
+-
+-   marshalSize = marshalSize + 4;  // _variableDatumID
+-   marshalSize = marshalSize + 4;  // _variableDatumLength
+-
+-   for(size_t idx=0; idx < _variableDatums.size(); idx++)

```

Comment:
This is a patch file which, according to the README.txt in the patches directory, is applied to generated code. 
Changing this delete line will corrupt the patch, the file that this would be applied to wouldn't have a matching line. 
Although, I wonder if the patches are needed any more? 
Will this repo be overwritten with generated code in future?

Reply:
The patches folder is really just historical evidence at this point. Years ago when the project was young the code was being regenerated often to fix broad problems. But now the generated code is about as good as it's ever going to get, and we're fixing minor issues directly on the resulting code. So no I don't expect to regenerate code in the future.